src/transformers/trainer.py

```diff
@@ -1674,7 +1675,11 @@ def _hp_search_setup(self, trial: Union["optuna.Trial", Dict[str, Any]]):
 
             self.args.hf_deepspeed_config = HfTrainerDeepSpeedConfig(self.args.deepspeed)
             self.args.hf_deepspeed_config.trainer_config_process(self.args)
-            self.args.deepspeed_plugin = DeepSpeedPlugin(hf_ds_config=self.args.hf_deepspeed_config)
+
+            # From 1.0 on, we also need to fully reset/override the DS plugin when doing sweeps
+            if AcceleratorState._shared_state != {} and version.parse(accelerate_version) > version.parse("0.34.2"):
+                self.args.deepspeed_plugin = None
+                AcceleratorState().deepspeed_plugins = DeepSpeedPlugin(hf_ds_config=self.args.hf_deepspeed_config)
```

Comment:
I think that the else condition is missing

Reply:
Prior to 1.0 it didn't matter at all, so the only new blocker is with 1.0.

Reply:
Post offline discussion, technically we should be able to do `AcceleratorState().reset_state(False)` here instead and it should do what we want without issue. Will try this alternative (then we don't need to limit it to a 1.0 flag either)